### ISSUE TITLE
Cache uploads when no network is available and upload once connected

### ISSUE THREAD

**Issue by eikek** [docspell/android-client#1]

**Comment by TheOneValen** [comment]
You would probably have to allow file access. But I am not sure, as the app should cache it in its own scope.

I Would go about this for every send, every send should be like this for the sake of less complexity:

* Cache file to send and add it to a queue
* A queue Worker tries to upload it and and dequeues it on success
* no Success -> no dequeue and retry with interval, increasing delay and stuff like that

**Comment by eikek** [comment]
Yes good idea, I also thought about it in a similar way. I thought about different queue impls to avoid requiring another permission. But I haven't really thought about it yet :-). To me this is not a high prio right now, because I don't currently need it.
